packages/generated-react/dist/eliza-BigIntService_connectquery_hooks.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// @generated by protoc-gen-connect-query-with-hooks v0.0.1 with parameter "target=ts,import-hook-from=@tanstack/react-query"
+// @generated from file eliza.proto (package buf.connect.demo.eliza.v1, syntax proto3)
+/* eslint-disable */
+// @ts-nocheck
+
+import { count } from "./eliza-BigIntService_connectquery.ts";
```

Comment:
> This import means that this plugin cannot work without the connect-query plugin also running.

I think this is trade-off. Because this plugin need to call `connect-query` under hood somehow, i think it's make our plugins be in a line together.

> since if someone specifies ONLY the hook version, imports will fail.
Yeah, that why I was put it in README to install both connect-query and hooks plugin

> I think it would be a huge advantage to not have to rely on that
@paul-sachs could you help suggest on this. I've no clue how to make it work without connect-query/ please help me :+1:

Reply:
@vctqs1, what Paul means is that the code generated by protoc-gen-connect-query-with-hooks requires three other plugins to work:

```yaml
plugins:
  - name: es
    out: dist
    opt:
      - target=ts
      - import_extension=.js
  - name: connect-web
    out: dist
    opt:
      - target=ts
      - import_extension=.js
  - name: connect-query
    opt: target=ts
    out: dist
  - name: connect-query-with-hooks
    opt: 
      - target=ts
      - import-hook-from=@tanstack/react-query
    out: dist
```

It would be a better user experience if the new plugin did not rely on code generated by protoc-gen-connect-query, so that you would simply use protoc-gen-connect-query if you want the standard behavior, protoc-gen-connect-query-with-hooks if you want the hooks, but not both at the same time.

The best way to do that is do duplicate code in the plugin where necessary. Instead of importing `count`, generate `count`.

Reply:
Yeah, you'd basically want to copy the content from [here](https://github.com/bufbuild/connect-query/blob/main/packages/protoc-gen-connect-query/src/generateTs.ts) and do roughly the same thing. You can probably move the jsDoc and export since that's not necessary, but you'll basically want to call `createQueryService` from `@bufbuild/connect-query` in your generated code.

Reply:
@timostamm I'd want to build ontop of this base to add those APIs. There are more streamlined apis that don't require `useQueryClient` calls all over the place. There are APIs like `invalidateQueries` which would be really helpful that `UnaryHooks` doesn't implement directly that could be done here. But an incremental approach makes sense to me.

Reply:
Thank you, I got you two idea.